packages/lib/src/createEventEmitter/createEventEmitter.ts

```diff
@@ -0,0 +1,40 @@
+import {InnerEventEmitter} from './createEventEmitter.types';
+
+export const createEventEmitter = () => {
```

Comment:
considering typings it's easier to write own impl for 40 rows, then trying to adopt some of solutions out there. Usually they are either untyped or require Record<'eventName', arguments> typings, which is leading to very complicated typings for `trigger` or passing some stuff like `undefined as any as never` as second argument when event doesn't pass anything to `trigger` because event have to be defined like 
```
type Events = {
 'someEvent:begin':  never;
 'someEvent:end: {errors: Errors}; 
}
```
TLDR:
it wasn't worth to fight these typings. Implementation is rather simple and straight-forward. Fully typed callback for every type of event.

Reply:
Found `strict-event-emitter`, which plays well with types
```
 type FormagusEvent = {
  'submit:begin': [],
  'submit:end': [Omit<SubmitParams, 'event'>],
}
  ```
and no issues with usage.
but it's pretty big and adds 5 kb more to bundle. I think, considering that this is pretty basic stuff - it's not worth it.

Reply:
Wow. Found `mitt`. It's super small and well typed. Gonna use it. The only downside is that own impl returned unsubscribe function from `on` call, and `mitt` doesn't do it, you have to call `off` with exact function passed as listener to remove specific listener (like unsubscribe for DOM events)

Reply:
see this commit 
https://github.com/iaroslavshvets/formagus/pull/105/commits/7f33595ff8222f385e58fc409e13c80a98585287